src/options/opt.cc

```diff
@@ -283,8 +283,8 @@ LOCAL_NODISCARD(Ret fix_mutopt(const conopt_t& glob,
         if (real.cgoto) {
             RET_FAIL(error("-g, --computed-gotos option is not supported for non-C backends"));
         }
-        if (real.case_ranges) {
-            RET_FAIL(error("--case-ranges option is not supported for non-C backends"));
+        if (glob.lang != Lang::D && real.case_ranges) {
+            RET_FAIL(error("--case-ranges option is not supported for non-C/D backends"));
```

Comment:
Move this out of the `if (glob.lang != Lang::C)`, and make the condition `if (real.case_ranges && !(glob.lang == Lang::C || glob.lang == Lang::D))` so that it is symmetrical wrt. C and D.

Reply:
Also, are you sure default (pointer) API cannot be supported with D? The ported examples suggest that it can, as you define generic API via pointers YYCURSOR, etc.